src/huggingface_hub/repository.py

```diff
@@ -730,6 +785,45 @@ def git_push(self) -> str:
 
         return self.git_head_commit_url()
 
+    def git_checkout(self, revision, create_branch_ok=False):
+        """
+        git checkout a given revision
+
+        Specifying `create_branch_ok` to `True` will create the branch to the given revision if that revision doesn't exist.
```

Comment:
I would put this in Args docstring for consistency with other methods (such as `push_to_hub`).

Reply:
All other methods in `Repository` are defined as this one is (except `push_to_hub` which has, unfortunately, a very uninformative description). I agree with you that `Args` is better, but I'd rather do it all in a single refactor. Does that sound good?

Reply:
Yes, that sounds great!